### PR TITLE
Upgrade react native 0.57

### PR DESCRIPTION
**Pull request by StevePotter**
Used `react-native-git-upgrade` and did some manual work to get everything upgraded to the latest version, including `react-native-background-upload`

Tested on Android and iOS simulators (no real devices) and it worked beautifully.

### CODE REVIEW

File: ios/ReactNativeBackgroundUploadExample/Info.plist [hunk 1]
```diff
@@ -49,6 +51,8 @@
 	<key>NSAppTransportSecurity</key>
 	<!--See http://ste.vn/2015/06/10/configuring-app-transport-security-ios-9-osx-10-11/ -->
 	<dict>
+    <key>NSAllowsArbitraryLoads</key>
```
Comment (TSMMark) [inline ios/ReactNativeBackgroundUploadExample/Info.plist]: mixed tabs and spaces

File: .flowconfig [hunk 2]
```diff
@@ -41,7 +57,5 @@ suppress_comment=\\(.\\|\n\\)*\\$FlowIssue\\((\\(>=0\\.\\(4[0-5]\\|[1-3][0-9]\\|
 suppress_comment=\\(.\\|\n\\)*\\$FlowFixedInNextDeploy
 suppress_comment=\\(.\\|\n\\)*\\$FlowExpectedError
 
-unsafe.enable_getters_and_setters=true
-
 [version]
-^0.45.0
+^0.78.0
```
Comment (TSMMark) [inline .flowconfig]: # 🚀

File: .nvmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+8.15.0
```
Comment (TSMMark) [inline .nvmrc]: 👍

File: android/app/proguard-rules.pro [hunk 4]
```diff
@@ -15,56 +15,3 @@
 #-keepclassmembers class fqcn.of.javascript.interface.for.webview {
 #   public *;
 #}
-
```
Comment (TSMMark) [inline android/app/proguard-rules.pro]: 👋 byebye

File: android/gradlew.bat [hunk 5]
```diff
@@ -46,10 +46,9 @@ echo location of your Java installation.
 goto fail
 
 :init
-@rem Get command-line arguments, handling Windowz variants
+@rem Get command-line arguments, handling Windows variants
```
Comment (TSMMark) [inline android/gradlew.bat]: LOL

File: android/settings.gradle [hunk 6]
```diff
@@ -4,4 +4,4 @@ project(':react-native-image-picker').projectDir = new File(rootProject.projectD
 include ':react-native-background-upload'
 project(':react-native-background-upload').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-background-upload/android')
 
-include ':app'
+include ':app'
```
Comment (TSMMark) [inline android/settings.gradle]: intentional?

**Review by TSMMark — APPROVED** [review]
👍

File: android/settings.gradle (hunk 6 above)
Comment (StevePotter) [reply android/settings.gradle]: Yes, added by `react-native-git-upgrade`.  That's how new projects have it.  Not sure why though

File: ios/ReactNativeBackgroundUploadExample/Info.plist (hunk 1 above)
Comment (StevePotter) [reply ios/ReactNativeBackgroundUploadExample/Info.plist]: interesting, I'll fix

**Pull request merged**